### PR TITLE
Support for python 3.7 for build and deploy

### PR DESCRIPTION
**Pull request by shahsmit14**
Support for python 3.7 for build and deploy
- Major dependent packages are kept the same
- Just making sure the code is compatible with Python 3.7 as Python 3.6 is reaching End of Life

### CODE REVIEW

**Comment by snazzyfox** [comment]
Should we also take this opportunity and add support for all other modern python3 versions?

**Comment by shahsmit14** [comment]
> Should we also take this opportunity and add support for all other modern python3 versions?

For the immediate support with Python 3.7, goal is to not make any major code changes due to tight timelines. But in Q1-2022 this repo is planned for major dependent package updates.

File: requirements.txt [hunk 1]
```diff
@@ -6,4 +6,5 @@ pandas==0.25.3
 pykalman==0.9.5
 scipy==1.2.3
 statsmodels==0.10.2
-scikit-learn==0.19.2
+scikit-learn==0.24.2
+decorator==5.1.0
```
Comment (sayanchk) [inline requirements.txt:10]: Why do we need to add decorator? Do we know where we need it?

Comment (shahsmit14) [reply requirements.txt:10]: bayesian_changepoint/offline_changepoint_detection needs it and our test were failing because of that.

**Review by sayanchk — APPROVED** [review]

**Pull request merged**